package.json

```diff
@@ -147,5 +147,7 @@
       "**/packages/**/*.test.js"
     ]
   },
-  "dependencies": {}
+  "dependencies": {
+    "lodash.debounce": "^4.0.8"
+  }
 }
```

Comment:
This should be in `@uppy/dashboard` dependencies probably?

Reply:
@arturi, good point.
I also saw self-written debounce() function used in another plugin (I didn't change it for lodash's version, because I wasn't 100% sure if there are interchangeable).
In general I feel like that's some function we may want to use in other plugins.
What do you think?

Reply:
> What do you think, should we just be adding lodash.debounce to every plugin separately when we need it?

Yes, since they are installed separately like `npm i @uppy/core @uppy/dashboard` — can’t use root-level package.json for that.

I also saw self-written debounce() function used in another plugin (I didn't change it for lodash's version, because I wasn't 100% sure if there are interchangeable).

We should ask @goto-bus-stop if their debounce https://github.com/transloadit/uppy/blob/854654b135eece671288fadf52c08e97ff220c59/packages/%40uppy/core/src/Plugin.js#L7-L24 can be replaced with lodash? I guess there was a reason for a custom version maybe?

Reply:
@arturi, understood, will move lodash.debounce to Dashboard's package.json #TODO.

Reply:
My "debounce" only works for many synchronous calls. eg when a plugin calls setState() 10 times in a row synchronously it makes sure that the `updateUI` function is only called once at the end (by scheduling a ["microtask"](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/) using `Promise.resolve()`).
It doesn't support custom delays or anything that lodash has, so most likely lodash.debounce should be used instead.